packages/richtext-lexical/src/features/upload/server/nodes/UploadNode.tsx

```diff
@@ -17,15 +17,26 @@ import { $applyNodeReplacement } from 'lexical'
 import * as React from 'react'
 
 export type UploadData<TUploadExtraFieldsData extends JsonObject = JsonObject> = {
-  [TCollectionSlug in CollectionSlug]: {
-    fields: TUploadExtraFieldsData
-    // Every lexical node that has sub-fields needs to have a unique ID. This is the ID of this upload node, not the ID of the linked upload document
-    id: string
-    relationTo: TCollectionSlug
-    // Value can be just the document ID, or the full, populated document
-    value: DataFromCollectionSlug<TCollectionSlug> | number | string
-  }
-}[CollectionSlug]
+  fields: TUploadExtraFieldsData
+  // Every lexical node that has sub-fields needs to have a unique ID. This is the ID of this upload node, not the ID of the linked upload document
+  id: string
+  relationTo: string
+  // Value can be just the document ID, or the full, populated document
+  value: number | string | TypedCollection
+}
+
+/**
+ * UploadData_P4 is a more precise type, and will replace UploadData in Payload v4.
+ * I am not encouraging users to use this type via tsdoc yet, in case we find more breaking changes to do prior to v4.
```

Comment:
Mark as `@internal` if this is the case.

Reply:
It's not actually needed because it's not exported by the package (UploadData is exported in packages/richtext-lexical/src/index.ts). I'm adding it anyway, it doesn't hurt 😁